url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/step_over.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package backend;
+
+import frontend.FrontEnd;
+import frontend.handlers;
+import java.awt.Color;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.StringTokenizer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.text.BadLocationException;
+
+/**
+ *
+ * @author Tushar
+ */
+public class step_over implements Runnable
+{
+
+    public static boolean barrelshiftpresent = false;
+    public static Long pos = new Long(0);
+    public static RandomAccessFile br = null;
+    public static Integer current_line;
+    public static int ocommon;
+    public static int otransfer;
+    public static int in_line, fi_line, f_int;
+    public Thread t;
+    public ScanFile ob;
+    public File fpath;
+    public static boolean ocall;
+    public static int initial_pos;
+    public static int end_pos;
+    public static int k = 0;
+    public static int onext;
+    public static int callEncounter;
+    public static int exitCondition = 0;
+    StepRun obj;
+    // private static boolean commentflag = false;
+
+    public step_over(File path, int ini_line, int fin_line, int f)
+      {
+        exitCondition = ScanFile.call_count;
+        t = new Thread(this, "stepover");
+        in_line = ini_line;
+        fi_line = fin_line;
+        f_int = f;
+
+        fpath = path;
+      }
+
+
+    
+    
+
+    @Override
+    public void run()
+      {
+        try
+        {
+            //System.out.println("step_out run()");
+            scanstart();
+        } catch (InterruptedException ex)
+        {
+            Logger.getLogger(step_out.class.getName()).log(Level.SEVERE, null, ex);
+        }
+      }
+
+    public static boolean branchToProcedure(String strLine) throws BadLocationException
+      {
+
+        if (new StringTokenizer(strLine, " ,\t").countTokens() == 2 && strLine.toUpperCase().contains("BL"))
+        {
+            //  frontend.FrontEnd.activepane.getHighlighter().addHighlight(initial_pos, end_pos, frontend.handlers.cyanPainter);
+            return true;
+        }
+        return false;
+      }
+
+   
+
+    public void scanstart() throws InterruptedException
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.